docs/FAQ.md

```diff
@@ -1,15 +1,126 @@
+[skinit-spec]: https://www.amd.com/system/files/TechDocs/24593.pdf
+
+### So, why TrenchBoot?
+
+- Managing multiple independent components individually can be confusing
+- Most DRTM systems are platform-dependent
+- Most DRTM systems can only measure entities already present in RAM at the
+    time DRTM sequence is initiated
+- Most DRTM systems support only one attestation action
+
+Trenchboot is meant to solve these, and many other problems by providing a
+universal framework for building integrity of software components in the
+process of booting a computer system.
+
+## 2. How does TrenchBoot work? (TrenchBoot architecture)
+
+### Architecture Overview
+
+*The full architecture overview can be found on
+[the corresponding page](./theory/Architecture.md).*
+![Overview](img/Architectural_Flow.png)
+This diagram shows a general overview of the overall architecture of a
+TrenchBoot system. In Bootstrap phase the system environment is measured and
+that information stored in the dynamic PCRs, using whatever method is
+appropriate for the given hardware. The intermediate phase analyzes the
+information collected in the bootstrap phase to decide if it wants to continue
+booting and pass control to the target OS, or if it wants to stop immediately.
+If desired, additional evidence/measurements can be gathered in this phase. Once
+the integrity of the system is verified the TrenchBoot kernel passes off
+control to the actual desired OS to initiate the runtime phase.
+
+### Bootstrap Phase - GRUB
+
+TrenchBoot utilizes its own system to carry out TXT on Intel, or SKINIT on AMD,
+which is a custom version of GRUB that has commands built in to carry out a
+TrenchBoot _Secure Launch_. Those commands are `slaunch` and `slaunch_module`.
+
+The custom GRUB works with:
+
+- Linux boot protocol (Intel only, legacy and UEFI),
+- Multiboot2 boot protocol (both Intel and AMD, but only legacy boot).
+
+Support for [AMD Linux](https://github.com/TrenchBoot/trenchboot-issues/milestone/8)
+and [Multiboot2 UEFI](https://github.com/TrenchBoot/trenchboot-issues/milestone/11)
+will be added later.
+
+### Intermediate Phase - Linux Kernel / U-ROOT (Initramfs)
+
+The intermediate phase processes information collected in the bootstrap phase
+and converts it into normalized data. This data is eventually used to determine
+whether to boot into the target OS. The kernel and u-root initramfs together
+make up TrenchBoot's _Security Engine_, which is an intermediate mini-OS that
+serves to process data gathered by that bootstrap phase. During the
+intermediate phase, the kernel and initramfs work together to measure block
+devices, individual files, SMBUS/DMI information and provides unseal based,
+external device and network based attestation. TrenchBoot calls this
+functionality secure launch (aka slaunch).
+
+## 3. Why does TrenchBoot use an intermediate launcher?
```

Comment:
https://github.com/TrenchBoot/TrenchBoot.github.io/pull/41#discussion_r1739744306

Reply:
The intermediate loader/launcher is just one use case, it just happens to be the use case that was implemented because, 1. it effectively mimics how tboot works allowing it to be a drop-in replacement, 2. it met the need of Oracle, and 3. provides components that can be reused to implement other use cases.

If this question is to stay, then it should read "What is the TrenchBoot intermediate loader?", and the answer should be updated that it tends to be the most common DRTM use-case, it's how tboot works and how Microsoft's Secure Core DRTM works.